server/sockets/socket-server.js

```diff
@@ -0,0 +1,74 @@
+const {io} = require('./../index')
+
+// on connecting with  client socket
+io.on('connection', (socket) => {
+  console.log('connected')
+  // join event listner
+  socket.once('join', function (roomname) {
+    // checks for maximum users and if game started?
+    if ((io.nsps['/'].adapter.rooms[roomname] && io.nsps['/'].adapter.rooms[roomname].length > 11) || (io.nsps['/'].adapter.rooms[roomname] && io.nsps['/'].adapter.rooms[roomname].allowed != undefined)) {
+    } else {
+      socket.join(roomname)      // if passed checks joins room
+      socket.broadcast.emit('broadcast', 'new player joined');      // brodcast every other a player has joined
+      socket.emit('connectToRoom', "You are in room " + roomname)   // emit user the roomname
+    }
+  })
+
+  // set username event listner
+  socket.on('setUsername', function (data) {
+    var usernameTaken = 0;                                   // var for checking unique username
+    io.in(data.roomname).clients((error, clients) => {      // getting all clients in room
+      if (error) throw error;
+      // loop for checking username of every other player with given username
+      for (var i = 0; i < clients.length; i++) {
+        var user = io.sockets.sockets[clients[i]];    // saving socket into user 
```

Comment:
Are you changing the user variable after assigning it? If not then it should be `const` instead of `var`

Reply:
I GUESS FOR EVERY VARIABLE ' i ' IT WOULD CHANGING CONTAINING DIFFERENT USER SOCKETS.

Reply:
maybe using let would be best because i am not using it anywhere else beside this block? should i change it to let ?

Reply:
you can use `const` here as remains constant during an iteration.

Reply:
okay changing it to const then

Reply:
DONE